api/src/main/java/javax/enterprise/inject/spi/ProcessInjectionPoint.java

```diff
@@ -56,6 +59,18 @@
     public void setInjectionPoint(InjectionPoint injectionPoint);
 
     /**
+     * Returns an {@link InjectionPointConfigurator} initialized with the {@link InjectionPoint} processed by this event
+     * to configure a new InjectionPoint that will replace the original one at the end of the observer invocation.
+     *
+     * Each call returns the same InjectionPointConfigurator
+     *
+     * @return a non reusable {@link InjectionPointConfigurator} to configure the replacing InjectionPoint
+     * @throws IllegalStateException if called outside of the observer method invocation
+     * @since 2.0
+     */
+    public InjectionPointConfigurator configureInjectionPoint();
```

Comment:
All similar methods should define what happens if a configurator is in invalid state, e.g. the user breaks something. But this should be probably defined for `setInjectionPoint()` and friends as well.